server/controllers/peopleController.ts

```diff
@@ -14,8 +16,14 @@ export default class PeopleController {
         try {
           const person = await this.personService.findByCrn(req.user.token, crn, !!checkCaseload)
           req.flash('crn', person.crn)
+
+          if (!isFullPerson(person)) {
+            throw new RestrictedPersonError(person.crn)
+          }
         } catch (err) {
-          if ('data' in err && err.status === 404) {
+          if (err.type === 'RESTRICTED_PERSON') {
+            addErrorMessageToFlash(req, err.message, 'crn')
```

Comment:
Ah, I think I understand - whether we get a full person or a restricted person depends on who the user is. So one person might be able to create an application for a given CRN because they have access to the full record, but another person might just see it in the dashboard and shouldn't be able to create an application for the person

Reply:
thanks, I'll see if I can make it a bit clearer